tests/integration_test/core_mqtt_system_test.c

```diff
@@ -246,6 +252,34 @@
  */
 #define MILLISECONDS_PER_TICK                      ( MILLISECONDS_PER_SECOND / configTICK_RATE_HZ )
 
+/**
+ * @brief Helper function-like macro to test for a specified incoming packet
+ * from the broker. This helper macro allows us to wait for the specified
+ * incoming packet without unnecessarily blocking on MQTT_ProcessLoop() for
+ * beyond its arrival.
+ *
+ * By using a macro instead of a function, the line number in the test case will
+ * be printed in a failure.
+ *
+ * @param[in] flag Packet flag to test for true.
+ * @param[in] expectedStatus Status code expected to be returned from MQTT_Process
+ */
+#define WAIT_FOR_PACKET( flag, expectedStatus )                                         \
```

Comment:
What advantage does this have over the function? Seems like the input-parameter types are always expected to be (bool, MQTTStatus_t).